proxy/http2/Http2ConnectionState.cc

```diff
@@ -898,17 +900,11 @@ Http2ConnectionState::main_event_handler(int event, void *edata)
                 error.msg);
         }
         this->send_goaway_frame(this->latest_streamid_in, error.code);
+        this->ua_session->set_half_close_flag(true);
```

Comment:
`set_half_close_flag` is changing the state of ua_session too.  Should we hold a mutex for it too?